### PR TITLE
XST and fiat currencies rates

### PR DESCRIPTION
**Pull request by Mingela**
### Description of the Change
Related to https://github.com/d3ledger/report-system/pull/53 
Provides unusual assets rates (calculated not by the curve)
Fiat assets will be added through a changelog

### CODE REVIEW

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt [hunk 1]
```diff
@@ -85,24 +90,53 @@ class ExchangerService(
...
+            }
+        }
+    }
+
+
+    /**
+     * Queries Iroha for trading pairs details and loads it
+     */
+    private fun updateTradingPairs() {
+        queryhelper.getAccountDetails(exchangerAccountId, tradePairSetter, tradePairKey).map {
```
Comment (dolgopolovwork) [inline exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt:120]: Errors are not handled

File: exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt [hunk 2]
```diff
@@ -85,24 +90,53 @@ class ExchangerService(
...
+                logger.info { "Updated trading pairs: $unEscape" }
+            }
+        }
+    }
+
+    /**
+     * Queries Iroha for unusual assets details and loads it
+     */
+    private fun updateUnusualAssets() {
+        queryhelper.getAccountDetails(exchangerAccountId, tradePairSetter, unusualAssetsKey).map {
```
Comment (dolgopolovwork) [inline exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt:143]: The same

File: exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt [hunk 3]
```diff
@@ -85,24 +90,53 @@ class ExchangerService(
...
-                    )
-                    logger.info { "Updated pairs: $unEscape" }
-                }
+            if (exchangerDetails.any { command -> command.key == unusualAssetsKey }) {
+                updateUnusualAssets()
+            }
+        }
+    }
+
+
```
Comment (dolgopolovwork) [inline exchanger/src/main/kotlin/com/d3/exchange/exchanger/ExchangerService.kt]: Extra space

File: exchanger/src/main/kotlin/com/d3/exchange/exchanger/RateStrategies.kt [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            } else {
+                getRateFor(from)
+            }
+        val toRate: BigDecimal =
+            if (to == baseAssetId) {
+                BigDecimal.ONE
+            } else {
+                getRateFor(to)
+            }
+        return fromRate.divide(toRate)
```
Comment (dolgopolovwork) [inline exchanger/src/main/kotlin/com/d3/exchange/exchanger/RateStrategies.kt:46]: Can `toRate` be zero?

Comment (Mingela) [reply exchanger/src/main/kotlin/com/d3/exchange/exchanger/RateStrategies.kt:46]: No, but if it happens an exception leading to a rollback will be thrown

**Review by dolgopolovwork — CHANGES_REQUESTED** [review]
Please handle errors properly

**Review by dolgopolovwork — APPROVED** [review]

**Pull request merged**